addons/web_editor/static/src/js/editor/add_snippet_dialog.js

```diff
@@ -113,27 +113,64 @@ export class AddSnippetDialog extends Component {
      * Inserts the snippets from the selected snippetGroup into the <iframe>.
      */
     async insertSnippets() {
-        let snippetsToDisplay = [];
-        const search = this.state.search.toLowerCase();
-        if (search) {
-            const strMatches = str => !search || str.toLowerCase().includes(search);
-            snippetsToDisplay = [...this.props.snippets.values()].filter(snippet => {
-                if (!snippet.excluded && ((snippet.group && !snippet.fromCategory)
-                    || ["structure", "hybrid"].includes(snippet.fromCategory))) {
-                    const matches = strMatches(snippet.category.text)
-                        || strMatches(snippet.displayName)
-                        || strMatches(snippet.data.oeKeywords || '');
-                    if (matches) {
-                        return snippet;
-                    }
-                }
+        // First, filter out snippets which are never supposed to be shown
+        // (excluded ones, inner content ones, ...).
+        let snippetsToDisplay = [...this.props.snippets.values()].filter(snippet => {
+            return !snippet.excluded
+                && (
+                    snippet.group && !snippet.fromCategory // Normal ones
+                    || ["structure", "hybrid"].includes(snippet.fromCategory) // Not-inner custom ones
```

Comment:
@qsm-odoo This line seems unnecessary; I don't understand why I introduced it because, in any case, the snippets that are not "inner content" don't have a group. All the others have one. And by the way, if that wasn't the case, the else without `this.state.search` wouldn't make sense because we're filtering only on `snippet.group` (after your changes) 🤔
So, either we can remove this line (which seems to be the solution based on my tests), or we need to revise the "else" if there is no `this.state.search`.

Reply:
Is not the group equal to "custom"? Otherwise how could my `else` have been working yesterday when clicking on the "custom" category?

Reply:
Yes,
- If the snippet is custom => group =  "custom"
- If the snippet is "inner content" (custom or not) => group = ""

So, I think `!snippet.excluded && snippet.group` is enough.
But we can leave it as it is for now and investigate later...